### DISCUSSION TITLE
Leverage the Gateway and Link functionalities for other endpoints besides Files

### DISCUSSION THREAD

**Discussion by grugna** [CBIIT/ccdi-federation-api#71] [API Design Discussion]
Initial Slack discussion [here](https://uc-cri.slack.com/archives/C05GL22240Y/p1706638406686269).

PCDC won't be able to share line level data beside the identifiers. The GET `/subject/{namespace}/{name}` will not return basically anything useful, and the GET `/subject` will have the IDs and the count. I envision using the Gateway and Link models to include information on how to access those data and where to find them.

I propose adding an "information" (any other preferred name is welcome)  attribute to the root endpoints in addition to "summary" and "data".
In our specific use case this is where I would add the gateway / link information. For instance our /subject endpoint will return the summary, the partial data containing only the identifiers, so I would need another attribute to contain the information for requesting access to the rest of the data.

```
{
  "summary": {
    "counts": {
      "total": 22328
    }
  },
  "data": [],
  "information": {
    "link": {
      "url": "https://portal.pedscommons.org/explorer",
      "instructions": "Select the correct filters and click on the request access button on the top right side of the screen.",
      "link_type": "Approximate"
    },
    "gateway_type": "Controlled"
  }
}
```
Note that the 'data' field is still empty here because we are still working on that piece, and the 'link.instructions' is just a blurb of text that needs to be refined. Eventually we will have a direct link to the data there.

**Comment by claymcleod** [comment]
Just thinking this through a little more—would you be opposed to making the top-level gateways key (that is available on files at present) as a cross-entity (subject/sample/file) key? In that way, we could essentially warrant to the community that this key represents any external link related to any entity in the API specification (including new ones that we add in the future)

**Reply by grugna** [reply]
I think it would be great to have that cross-entity.

**Reply by e-t-k** [reply]
as discussed in the meeting just now, I also agree with making this a cross-entity.

**Comment by claymcleod** [comment]
In the meeting today, every federation member approved my above, proposed change addressing this. I will add this in a future iteration.
